lib/idris-ide-mode.coffee

```diff
@@ -9,15 +9,16 @@ class IdrisIdeMode extends EventEmitter
   buffer: ''
   idrisBuffers: 0
 
-  constructor: ->
-    pathToIdris = atom.config.get("language-idris.pathToIdris")
-    @process = spawn pathToIdris, ['--ide-mode']
-    @process.on 'error', @error
-    @process.on 'exit', @exited
-    @process.on 'close', @exited
-    @process.on 'disconnect', @exited
+  start: ->
+    if (not @process?) || @process.killed
+      pathToIdris = atom.config.get("language-idris.pathToIdris")
+      @process = spawn pathToIdris, ['--ide-mode']
+      @process.on 'error', @error
+      @process.on 'exit', @exited
```

Comment:
No, good point. Not entirely sure what would be the most elegant way to do this.
Maybe converting the callbacks to RxJS observables and using something like this: https://github.com/Reactive-Extensions/RxJS/blob/master/doc/api/core/operators/debounce.md

I don't like the callbacks anyway